storage_controller/src/reconciler.rs

```diff
@@ -811,6 +811,11 @@ impl Reconciler {
                     tenant_conf: self.config.clone(),
                 },
             ));
+
+            // TODO: Consider notifying control plane about detaches. This would avoid situations
+            // where the compute tries to start-up with a stale set of pageservers.
+            self.compute_hook
```

Comment:
In this code path, we are usually detaching from _this_ pageserver (it is referenced in observed but not intent) but the shard is still attached to some other pageserver.

We should only be clearing our compute notify state if the whole tenant is configured with a Detached policy.  Could infer that in Reconciler by inspecting intent.attached?

Reply:
> We should only be clearing our compute notify state if the whole tenant is configured with a Detached policy. 

Hm. This wasn't my intention. In theory you can have two concurrent reconciles for shards of the same tenant where one detaches and another one attaches. If the detach happens before the attach compute notification, then we would send compute notification containing a stale pageserver. This could happen when the tenant policy is updated to detach and there's one or more on-going reconciles for it's shards.

Reply:
> but the shard is still attached to some other pageserver

This is a good point though. We should only clear when intent attached is None.